connectors/utils.py

```diff
@@ -653,3 +657,112 @@ async def aenumerate(asequence, start=0):
             yield i, elem
         finally:
             i += 1
+
+
+class ExtractionService:
```

Comment:
should we have that in its own module?

Reply:
I want it to be but we discussed in the last projects sync that we should just put it in utils for now.